src/app/server/Server.h

```diff
@@ -98,8 +100,15 @@ class Server
     {
         CHIP_ERROR SyncGetKeyValue(const char * key, void * buffer, uint16_t & size) override
         {
-            ReturnErrorOnFailure(DeviceLayer::PersistedStorage::KeyValueStoreMgr().Get(key, buffer, size));
+            size_t bytesRead;
+            ReturnErrorOnFailure(DeviceLayer::PersistedStorage::KeyValueStoreMgr().Get(key, buffer, size, &bytesRead));
+            if (!CanCastTo<uint16_t>(bytesRead))
+            {
+                ChipLogDetail(AppServer, "%zu is too big to fit in uint16_t", bytesRead);
```

Comment:
%zu is not available on all embedded platforms and may cause unexpected operation where unavailable. Suggest using a fixed with integer like uint16_t/32_t and PRIu16/32

Reply:
Addressed in https://github.com/project-chip/connectedhomeip/pull/13600